src/graph.rs

```diff
@@ -27,27 +29,55 @@ impl GitGraph {
             repository,
             commits,
         };
-        graph.assign_branches(indices)?;
+        graph.assign_branches(indices, settings)?;
 
         Ok(graph)
     }
-    fn assign_branches(&mut self, indices: HashMap<Oid, usize>) -> Result<(), Error> {
-        let branches = self.repository.branches(None)?;
-        for branch in branches {
-            let (branch, branch_type) = branch?;
-            if branch_type == BranchType::Local {
-                let reference = branch.get();
-                if let Some(name) = reference.name() {
-                    if let Some(oid) = reference.target() {
-                        let idx = indices[&oid];
-                        self.commits[idx].branches.push(name[11..].to_string());
-                    }
+    fn assign_branches(
+        &mut self,
+        indices: HashMap<Oid, usize>,
+        settings: &Settings,
+    ) -> Result<(), Error> {
+        let branches_ordered = Self::branches_persistence_order(&self.repository, settings)?;
+
+        for branch in branches_ordered {
+            let reference = branch.get();
+            if let Some(name) = reference.name() {
+                if let Some(oid) = reference.target() {
+                    let idx = indices[&oid];
+                    self.commits[idx].branches.push(name[11..].to_string());
                 }
             }
         }
 
         Ok(())
     }
+
+    fn branches_persistence_order<'repo>(
```

Comment:
Why make this an associated function at all instead of just a free function at the module scope?

Reply:
Good question! Probably just a habit from languages where this is not possible.